### PR TITLE
Add integration framework

### LINKED ISSUE #1050: Create entry point for "Integration Framework" documentation in the Modeler component sections
With the introduction of the [connector template editor in the Web Modeler](https://github.com/camunda/product-hub/issues/64), element templates are no longer a Desktop Modeler - only feature. Therefore, we need to document it in the scope of the Web Modeler, too.

Once we retrieve the [latest element template schema](https://github.com/bpmn-io/bpmn-js-properties-panel/issues/690) in the Web Modeler that adds support for conditional templates, we need to move the [element template docs](https://docs.camunda.io/docs/components/modeler/desktop-modeler/element-templates/defining-templates/) out of the Desktop Modeler sections into a generic one and reference it from the Web Modeler sections (https://docs.camunda.io/docs/components/modeler/web-modeler/connectors/) (there is already a reference into the Desktop documentation that needs to be updated). 

This can be challenging due to the "Element Template"/"Connector Template" name clash, so I prefer naming the new common section "Integration Framework" and place it into the docs like this (note it is just a mockup)

image

and have the 2 dedicated sections for both Web and Desktop documentation refer to it, but with additional, product-specific details (i.e. how to include them in Desktop vs. how to write them in Web).

And we should also remove this [notice](https://docs.camunda.io/docs/components/modeler/desktop-modeler/element-templates/about-templates/), because it is now part of the product we sell and thus public+reliable:

image

### PR DESCRIPTION
**Pull request by MaxTru**
closes #1050

This moves Connectors to its own space and introduce the terms:

- **integration framework**
- **connector SDK**
- **connector template**

Screenshot_20220809_104526

### CODE REVIEW

**Comment by MaxTru** [comment]
This is still draft, because I still need to:

- [ ] verify old links dont break / update routes
- [ ] wait for SDK release

Still adding reviewers:
* @PreetiNirwal could you please check whether this is what is envisioned from PM perspective?
* @akeller and @nikku for info

**Review by PreetiNirwal — APPROVED** [review]
Looks good, thanks!

**Comment by MaxTru** [comment]
FYI @CatalinaMoisuc and @christian-konrad this moves the connectors out of the web modeler directory

**Comment by MaxTru** [comment]
@nikku or @menski could one of you please double-check from engineering side? Thanks

**Comment by akeller** [comment]
@pepopowitz FYI - this PR will likely need some guidance after you merge #1118. Can you follow up with them once your PR is merged?

**Comment by pepopowitz** [comment]
Hi @MaxTru! I've just merged #1118. Unfortunately, that means this PR is no longer updating the docs version that it was originally. 

Changes to the `/docs/` folder now apply to the "next" unreleased version. If you'd like to update the docs for the "current" version (8.0), you'll need to apply these same changes to the `versioned_docs/version-8.0` folder.

I'm sorry for the inconvenience and double work. Let me know if I can answer questions or offer any assistance.

**Comment by MaxTru** [comment]
> Hi @MaxTru! I've just merged #1118. Unfortunately, that means this PR is no longer updating the docs version that it was originally.
> 
> Changes to the `/docs/` folder now apply to the "next" unreleased version. If you'd like to update the docs for the "current" version (8.0), you'll need to apply these same changes to the `versioned_docs/version-8.0` folder.
> 
> I'm sorry for the inconvenience and double work. Let me know if I can answer questions or offer any assistance.

Hi,

okay - so I did the changes now in 2 places: `docs` and `versioned-docs`. Can you please have a look and confirm?

Thanks

**Review by menski — DISMISSED** [review]
Thanks for the changes :cookie: 

FYI [Connectors SDK 0.1.0](https://repo1.maven.org/maven2/io/camunda/connector/connector-core/0.1.0/) is release, and the [repository](https://github.com/camunda/connector-sdk) is public; also the [connector template](https://github.com/camunda/connector-template) repository is updated and public

**Comment by MaxTru** [comment]
@akeller this is ready and it would be great to have this mergend released. Once we have this, we can share with partners.
However, there is no fixed deadline.

Could you please give an ETA when this would be public?

Thanks
Max

**Comment by nikku** [comment]
This is such a huge and important change. Thanks for your work on this @MaxTru !

File: docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md
Comment (nikku) [inline docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md:8]:
```suggestion
The **connector SDK** is in developer preview and subject to breaking changes. Use at your own risk!
```

File: versioned_docs/version-8.0/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md
Comment (nikku) [inline versioned_docs/version-8.0/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md:8]:
```suggestion
The **connector SDK** is in developer preview and subject to breaking changes. Use at your own risk!
```

**Comment by akeller** [comment]
> @akeller this is ready and it would be great to have this mergend & released. Once we have this, we can share with partners. However, there is no fixed deadline.
> 
> Could you please give an ETA when this would be public?
> 
> Thanks Max

@pepopowitz @christinaausley can you step through this PR and review based on the new changes to docs versioning? @pepopowitz I leave it up to you to decide when you want to merge & release given the items you are working through this week.

File: docs/components/components-overview.md
Comment (pepopowitz) [inline docs/components/components-overview.md:18]: I think this line is not also included in the 8.0 docs. Is that intentional?

**Review by pepopowitz — DISMISSED** [review]
I noticed one change from the `docs` folder that didn't also make it into the `version-8.0` folder. Aside from that, this looks exactly as I'd expect it to in regards to the next/8.0 changes!

**Comment by pepopowitz** [comment]
> Could you please give an ETA when this would be public?

Once this is merged, it should be no problem for me to deploy to production within a day.

File: docs/components/components-overview.md
Comment (MaxTru) [reply docs/components/components-overview.md:18]: ups - true. Thanks. Corrected

**Comment by MaxTru** [comment]
Hi @pepopowitz ,

I corrected your finding and enabled "auto-merge". Could you please approve? It should then be merged once the CI finished. Thanks

**Review by pepopowitz — DISMISSED** [review]

**Comment by akeller** [comment]
We still need a docs review for style. I've assigned @christinaausley as a reviewer.

**Review by christinaausley — DISMISSED** [review]
See my latest commit for details on technical review, including:
- Renaming doc ID to something more specific.
- Writing out Camunda Platform 8 rather than Camunda 8.
- Uppercase Self-Managed

**Review by akeller — APPROVED** [review]

**Comment by akeller** [comment]
@pepopowitz are you still around to merge and release?

**Comment by akeller** [comment]
Out with https://github.com/camunda/camunda-platform-docs/releases/tag/8.0.24 🚀

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md
Comment (christian-konrad) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: @MaxTru I am really confused by the naming. I thought connectors where a product name, thus starting with a capital letter, while secrets is something general, thus lowercase. But from this docs, it seems like it is the other way around.

I am asking since I am writing the docs for the Connector Template Editor and want to be consistent. What is the reasoning for this style here?

Comment (PreetiNirwal) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: I believe this is just a typo, oversight. We can update and capitalise Connectors, and lower case Secrets. Thanks.

Comment (akeller) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: @christinaausley, maybe you can help here? Or maybe this is a misunderstanding between what is a product name and what is not.

Comment (christinaausley) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: Agreed here that secrets should be lowercase.

As far as "connectors" goes -- this is purely up to preference. All of our products (Camunda Platform 8, Camunda Platform 7, Self-Managed) are capitalized, as are our key components (Optimize, Operate, etc.) Think of this like "Microsoft Excel" or "Microsoft Word" -- they are all key applications of a company, hence capitalization.

If we create concepts that are *supplemental* to our key components (think Optimize's "reports" and "dashboards" or "clusters", these are lowercase.

Thus, why I circle back to preference. Do we consider connectors supplemental to Modeler, or their own, independent product?

Comment (MaxTru) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: +1 for `secrets` and `Connectors`

Comment (christian-konrad) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: Ok I see that we do not capitalize `Connectors` on our websites. We only capitalize  `Integration Framework`. I will stick to that https://camunda.com/platform/modeler/connectors/

Comment (christinaausley) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: Great! I'll ensure this is mirrored in the [PR I've launched](https://github.com/camunda/camunda-platform-docs/pull/1174). @PreetiNirwal -- feel free to take a look at the files changed for a closer look at all of my changes.

Also note I've launched a [PR to lowercase secrets](https://github.com/camunda/camunda-platform-docs/pull/1175) and will launch a subsequent PR to ensure all instances of Connectors are uppercase.

Comment (akeller) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:13]: Pulling all the threads together. Let's follow up on this Slack thread with next steps - https://camunda.slack.com/archives/C026U8GBNSW/p1660769206379089

**Pull request merged**
